modules/tensor_mechanics/src/actions/DynamicTensorMechanicsAction.C

```diff
@@ -18,6 +18,7 @@ validParams<DynamicTensorMechanicsAction>()
   params.addParam<Real>("zeta", 0, "zeta parameter for the Rayleigh damping");
   params.addParam<Real>("alpha", 0, "alpha parameter for HHT time integration");
   params.addParam<bool>("static_initialization", false, "Set to true get the system to equillibrium under gravity by running a quasi-static analysis (by solving Ku = F) in the first time step.");
+  params.addParam<bool>("material_rayleigh_damping", false, "Set to true to apply material dependent rayleigh damping or rayleigh damping that varies with quadrature point.");
```

Comment:
Rather than adding the "material_rayleigh_damping" parameter, you can just convert the existing "zeta" parameter to a MaterialProperty:

params.addParam<MaterialPropertyName>("zeta", 0, "zeta parameter for the Rayleigh damping");
_zeta(getMaterialProperty<Real>("zeta))

If you do this, the user can either specify a number (it would default to 0) or the name of a material property.  That would simplify a number of aspects of this code, plus the user wouldn't have to guess about the name to use for the material property that supplies zeta.